api/v1alpha1/bundledeployment_types.go

```diff
@@ -49,6 +50,9 @@ type BundleDeploymentSpec struct {
 	ProvisionerClassName string `json:"provisionerClassName"`
 	// Template describes the generated Bundle that this deployment will manage.
 	Template *BundleTemplate `json:"template"`
+	// Config is provisioner specific configurations
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Config runtime.RawExtension `json:"config,omitempty"`
```

Comment:
Is this specifically for the helm provisioner? If so, I think we should name it to reflect that?